### PR TITLE
Adding support for bootstrap file

### PR DESCRIPTION
**Pull request by chenyanlann**
## What is the purpose of the change

Adding support for bootstrap file. 

## Brief changelog

When istio-agent start, a bootstrap file will be created, which contains the following message.

{
  "xds_servers": [
  ],
  "node": {
  }
}

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [x] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

File: dubbo-xds/src/main/java/org/apache/dubbo/registry/xds/XdsInitializationException.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package org.apache.dubbo.registry.xds;
```
Comment (wangchengming666) [inline dubbo-xds/src/main/java/org/apache/dubbo/registry/xds/XdsInitializationException.java:1]: add apache license

**Review by wangchengming666 — CHANGES_REQUESTED** [review]

File: dubbo-xds/src/main/java/org/apache/dubbo/registry/xds/util/bootstrap/BootstrapperImpl.java [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+        }
+        if (fileContent == null) throw new XdsInitializationException("Cannot find bootstrap configuration");
+
+        Map<String, ?> rawBootstrap;
+        try {
+            rawBootstrap = (Map<String, ?>) JsonParser.parse(fileContent);
+        } catch (IOException e) {
+            throw new XdsInitializationException("Failed to parse JSON", e);
+        }
+        return bootstrap(rawBootstrap);
```
Comment (wangchengming666) [inline dubbo-xds/src/main/java/org/apache/dubbo/registry/xds/util/bootstrap/BootstrapperImpl.java:50]: 或许你可以在这里对接XdsServiceDiscoveryFactory，然后你看下这个createDiscovery这个方法，里面的doInitialize

File: dubbo-xds/src/main/java/org/apache/dubbo/registry/xds/util/bootstrap/BootstrapperImpl.java [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+    }
+
+    @Override
+    BootstrapInfo bootstrap(Map<String, ?> rawData) throws XdsInitializationException {
+        BootstrapInfo.Builder builder = new BootstrapInfoImpl.Builder();
+
+        List<?> rawServerConfigs = JsonUtil.getList(rawData, "xds_servers");
```
Comment (wangchengming666) [inline dubbo-xds/src/main/java/org/apache/dubbo/registry/xds/util/bootstrap/BootstrapperImpl.java:73]: 这里用了很多的`?`，是因为类型都是未知的吗

File: dubbo-xds/src/test/java/org/apache/dubbo/registry/xds/util/bootstrap/BootstrapperTest.java [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+import org.apache.dubbo.registry.xds.XdsInitializationException;
+import org.hamcrest.MatcherAssert;
+
+import java.io.IOException;
+import java.util.List;
+
+import static org.apache.dubbo.registry.xds.util.bootstrap.BootstrapperImpl.BOOTSTRAP_PATH_SYS_ENV_VAR;
+import static org.hamcrest.Matchers.equalTo;
+
+public class BootstrapperTest {
```
Comment (wangchengming666) [inline dubbo-xds/src/test/java/org/apache/dubbo/registry/xds/util/bootstrap/BootstrapperTest.java:12]: use junit

File: dubbo-xds/src/test/java/org/apache/dubbo/registry/xds/util/bootstrap/BootstrapperTest.java [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+            "    }\n" +
+            "  },\n" +
+            "  \"server_listener_resource_name_template\": \"xds.istio.io/grpc/lds/inbound/%s\"\n" +
+            "}";
+        BootstrapperImpl.bootstrapPathFromEnvVar = "";
+        BootstrapperImpl bootstrapper = new BootstrapperImpl();
+        bootstrapper.setFileReader(createFileReader(rawData));
+        Bootstrapper.BootstrapInfo info = bootstrapper.bootstrap();
+        List<Bootstrapper.ServerInfo> serverInfoList = info.servers();
+        MatcherAssert.assertThat(serverInfoList.get(0).target(), equalTo("unix:///etc/istio/proxy/XDS"));
```
Comment (wangchengming666) [inline dubbo-xds/src/test/java/org/apache/dubbo/registry/xds/util/bootstrap/BootstrapperTest.java:112]: use Assertions.assertEquals

File: dubbo-xds/pom.xml [hunk 6]
```diff
@@ -84,6 +84,18 @@
             <groupId>org.bouncycastle</groupId>
             <artifactId>bcprov-ext-jdk15on</artifactId>
         </dependency>
+        <dependency>
```
Comment (wangchengming666) [inline dubbo-xds/pom.xml:87]: in dubbo-parent has org.junit.jupiter

**Review by wangchengming666 — CHANGES_REQUESTED** [review]

File: dubbo-xds/pom.xml (hunk 6 above)
Comment (chenyanlann) [reply dubbo-xds/pom.xml]: Thank you, chengming, I have changed this.

File: dubbo-xds/src/test/java/org/apache/dubbo/registry/xds/util/bootstrap/BootstrapperTest.java (hunk 5 above)
Comment (chenyanlann) [reply dubbo-xds/src/test/java/org/apache/dubbo/registry/xds/util/bootstrap/BootstrapperTest.java]: Thank you, chengming, I have involved Junit.

File: dubbo-xds/src/main/java/org/apache/dubbo/registry/xds/XdsInitializationException.java (hunk 1 above)
Comment (chenyanlann) [reply dubbo-xds/src/main/java/org/apache/dubbo/registry/xds/XdsInitializationException.java:17]: Thank you for your remind, chengming.

**Pull request merged**